extensions/wsc/src/main/java/org/mule/extension/ws/internal/ConsumeOperation.java

```
@@ -0,0 +1,165 @@
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+
+import net.sf.saxon.jaxp.SaxonTransformerFactory;
+import org.apache.cxf.attachment.AttachmentImpl;
+import org.apache.cxf.binding.soap.SoapHeader;
+import org.apache.cxf.endpoint.Client;
+import org.apache.cxf.message.Attachment;
+import org.apache.cxf.message.Exchange;
+import org.apache.cxf.message.ExchangeImpl;
+import org.w3c.dom.Element;
+import org.xml.sax.InputSource;
+
+/**
+ * The only {@link WebServiceConsumer} operation which consumes an operation of the connected web service and returns it's
+ * response.
+ * <p>
+ * The consume operation expects an XML body and a set of headers and attachments if required.
+ * <p>
+ * For the cases where no input parameters are required the {@link RequestBodyGenerator} will generate a body to perform the
+ * operation if a {@code null} value is passed.
+ *
+ * @since 4.0
+ */
+public class ConsumeOperation {
+
+  public static final String MULE_ATTACHMENTS_KEY = "mule.wsc.attachments";
+  public static final String MULE_HEADERS_KEY = "mule.wsc.headers";
+  public static final String MULE_SOAP_ACTION = "mule.wsc.soap.action";
+
+  @OutputResolver(output = OutputBodyResolver.class, attributes = WscAttributesResolver.class)
+  public OperationResult<String, WscAttributes> consume(@Connection WscConnection connection,
+                                                        @MetadataKeyId(OperationKeysResolver.class) String operation,
+                                                        @Optional @Content @TypeResolver(InputBodyResolver.class) String body,
+                                                        @Optional @TypeResolver(InputHeadersResolver.class) Map<String, String> headers,
+                                                        @Optional List<WsAttachment> attachments)
+      throws Exception {
```

we could throw an exception more in the WSC domain instead of just `Exception`. 
Is OnException something we could benefit of?